### PR TITLE
Add diskimage-builder workflow

### PR DESCRIPTION
**Pull request by Sunnatillo**

### CODE REVIEW

File: ci/dib_image/README.md
Comment (tuminoid) [inline ci/dib_image/README.md:1]: Can you format the markdown lines ot 80 chars and fix the bare url as well, they'll be caught by the new markdownlint soon.

File: ci/dib_image/build-centos-dev-image.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+set -eux
+
+export IMAGE_OS="centos"
+
+# Install disk-image-builder
+sudo dnf install python3 pip -y
+sudo pip install diskimage-builder
+
+source dib-and-image-vars.sh
```
Comment (tuminoid) [inline ci/dib_image/build-centos-dev-image.sh:11]:
```suggestion
. dib-and-image-vars.sh
```
`source` is deprecated, use `.` for the same.

File: ci/dib_image/build-centos-dev-image.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+# Create an image
+disk-image-create -a amd64 centos-dev centos -o ${IMAGE_NAME} block-device-mbr
+
+# Install openstackclient
+sudo pip3 install python-openstackclient
+
+source openstack-vars.sh
```
Comment (tuminoid) [inline ci/dib_image/build-centos-dev-image.sh:19]: Same.

File: ci/dib_image/build-centos-dev-image.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+export IMAGE_OS="centos"
+
+# Install disk-image-builder
+sudo dnf install python3 pip -y
+sudo pip install diskimage-builder
+
+source dib-and-image-vars.sh
+
+# Create an image
+disk-image-create -a amd64 centos-dev centos -o ${IMAGE_NAME} block-device-mbr
```
Comment (tuminoid) [inline ci/dib_image/build-centos-dev-image.sh:14]:
```suggestion
disk-image-create -a amd64 centos-dev centos -o "${IMAGE_NAME}" block-device-mbr
```

File: ci/dib_image/build-ubuntu-dev-image.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+set -eux
+
+export IMAGE_OS="ubuntu"
+
+# Install disk-image-builder
+sudo apt-get install python3-pip -y
+sudo pip install diskimage-builder
+
+source dib-and-image-vars.sh
```
Comment (tuminoid) [inline ci/dib_image/build-ubuntu-dev-image.sh:11]: Same

File: ci/dib_image/build-ubuntu-dev-image.sh [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+# Create an image
+disk-image-create -a amd64 ubuntu-dev ubuntu -o ${IMAGE_NAME} block-device-mbr
```
Comment (tuminoid) [inline ci/dib_image/build-ubuntu-dev-image.sh:14]:
```suggestion
disk-image-create -a amd64 ubuntu-dev ubuntu -o "${IMAGE_NAME}" block-device-mbr
```

File: ci/dib_image/build-ubuntu-dev-image.sh [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+# Install openstackclient
+sudo pip3 install python-openstackclient
+
+source openstack-vars.sh
```
Comment (tuminoid) [inline ci/dib_image/build-ubuntu-dev-image.sh:19]: Same

File: ci/dib_image/build-ubuntu-dev-image.sh [hunk 7]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
Comment (tuminoid) [inline ci/dib_image/build-ubuntu-dev-image.sh:1]:
```suggestion
#!/usr/bin/env bash
```

File: ci/dib_image/build-centos-dev-image.sh [hunk 8]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
```
Comment (tuminoid) [inline ci/dib_image/build-centos-dev-image.sh:1]:
```suggestion
#!/usr/bin/env bash
```

File: ci/dib_image/dib-and-image-vars.sh [hunk 9]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
```
Comment (tuminoid) [inline ci/dib_image/dib-and-image-vars.sh:1]:
```suggestion
#!/usr/bin/env bash
```

File: ci/dib_image/dib-and-image-vars.sh [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+
+set -eux
+
+current_dir="$(dirname "$(readlink -f "${0}")")"
```
Comment (tuminoid) [inline ci/dib_image/dib-and-image-vars.sh:5]:
```suggestion
CURRENT_DIR="$(dirname "$(readlink -f "${0}")")"
```
Rule of thumb: Global vars in UPPER case, local variables lower case.

File: ci/dib_image/dib_elements/centos-dev/README.md [hunk 11]
```diff
@@ -0,0 +1,12 @@
+# centos-dev element
+
+### Overview
+
+This element installs packages and changes configuration specifically for centos-dev images
```
Comment (tuminoid) [inline ci/dib_image/dib_elements/centos-dev/README.md:5]: Line length.

File: ci/dib_image/dib_elements/centos-dev/README.md [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+
+### Depends
+
+* centos
```
Comment (tuminoid) [inline ci/dib_image/dib_elements/centos-dev/README.md:9]: We use `-` for unordered lists.

File: ci/dib_image/dib_elements/centos-dev/element-deps [hunk 13]
```diff
@@ -0,0 +1,3 @@
+centos
+dev-base
+openssh-server
```
Comment (tuminoid) [inline ci/dib_image/dib_elements/centos-dev/element-deps:3]: Missing linefeed.

File: ci/dib_image/dib_elements/centos-dev/post-install.d/01_install [hunk 14]
```diff
@@ -0,0 +1,17 @@
+#! /usr/bin/env bash
```
Comment (tuminoid) [inline ci/dib_image/dib_elements/centos-dev/post-install.d/01_install:1]:
```suggestion
#!/usr/bin/env bash
```

File: ci/dib_image/dib_elements/centos-dev/post-install.d/02_configure [hunk 15]
```diff
@@ -0,0 +1,33 @@
+#! /usr/bin/env bash
```
Comment (tuminoid) [inline ci/dib_image/dib_elements/centos-dev/post-install.d/02_configure:1]:
```suggestion
#!/usr/bin/env bash
```

File: ci/dib_image/dib_elements/dev-base/README.md [hunk 16]
```diff
@@ -0,0 +1,14 @@
+# dev-base element
+
+### Overview
+
```
Comment (tuminoid) [inline ci/dib_image/dib_elements/dev-base/README.md:4]: Line length

File: ci/dib_image/dib_elements/dev-base/README.md [hunk 17]
```diff
@@ -0,0 +1,14 @@
+# dev-base element
+
+### Overview
```
Comment (tuminoid) [inline ci/dib_image/dib_elements/dev-base/README.md:3]:
```suggestion
## Overview
```
After `#` (1st level header) comes `##` (2nd level header), cannot jump directly to `###` (3rd level)

File: ci/dib_image/dib_elements/dev-base/README.md [hunk 18]
```diff
@@ -0,0 +1,14 @@
...
+
+This element takes care of installing common packages both for ubuntu and centos dev images
+
+### Depends
```
Comment (tuminoid) [inline ci/dib_image/dib_elements/dev-base/README.md:7]:
```suggestion
## Depends
```

File: ci/dib_image/dib_elements/dev-base/README.md [hunk 19]
```diff
@@ -0,0 +1,14 @@
...
+
+* base
```
Comment (tuminoid) [inline ci/dib_image/dib_elements/dev-base/README.md:9]:
```suggestion
- base
```
As before

File: ci/dib_image/dib_elements/dev-base/element-deps [hunk 20]
```diff
@@ -0,0 +1,6 @@
+base
+vm
+devuser
+openssh-server
+pkg-map
+package-installs
```
Comment (tuminoid) [inline ci/dib_image/dib_elements/dev-base/element-deps:6]: missing new line

File: ci/dib_image/dib_elements/ubuntu-dev/README.md [hunk 21]
```diff
@@ -0,0 +1,13 @@
+# ubuntu-dev element
```
Comment (tuminoid) [inline ci/dib_image/dib_elements/ubuntu-dev/README.md:1]: All same comments a previous file

File: ci/dib_image/dib_elements/ubuntu-dev/element-deps [hunk 22]
```diff
@@ -0,0 +1,3 @@
+ubuntu
+dev-base
+openssh-server
```
Comment (tuminoid) [inline ci/dib_image/dib_elements/ubuntu-dev/element-deps:3]: Line feed.

File: ci/dib_image/dib_elements/ubuntu-dev/post-install.d/01_install [hunk 23]
```diff
@@ -0,0 +1,43 @@
+#! /usr/bin/env bash
```
Comment (tuminoid) [inline ci/dib_image/dib_elements/ubuntu-dev/post-install.d/01_install:1]:
```suggestion
#!/usr/bin/env bash
```

File: ci/dib_image/dib_elements/ubuntu-dev/post-install.d/01_install [hunk 24]
```diff
@@ -0,0 +1,43 @@
...
+  "deb [arch=amd64 signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
+  $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list
+
+sudo apt-get update
+sudo apt-get install -y docker-ce docker-ce-cli containerd.io jq
+sudo groupadd docker
+sudo usermod -aG docker ubuntu
+sudo usermod -aG docker metal3ci
+sudo systemctl enable docker
+sudo systemctl restart docker
```
Comment (tuminoid) [inline ci/dib_image/dib_elements/ubuntu-dev/post-install.d/01_install:43]: Missing Line feed.

File: ci/dib_image/dib_elements/ubuntu-dev/post-install.d/02_configure [hunk 25]
```diff
@@ -0,0 +1,166 @@
+#! /usr/bin/env bash
```
Comment (tuminoid) [inline ci/dib_image/dib_elements/ubuntu-dev/post-install.d/02_configure:1]:
```suggestion
#!/usr/bin/env bash
```

File: ci/dib_image/dib_elements/ubuntu-dev/post-install.d/02_configure [hunk 26]
```diff
@@ -0,0 +1,166 @@
...
+sudo apt-get -y --purge remove telnet
+sudo apt-get -y autoremove
+
+# We do not use passwords on the machines
+
+# Disable the system accounts for non-root users
+
+# shellcheck disable=SC2013
+for user in $(awk -F: '($3 < 500) {print $1 }' /etc/passwd); do
+  if [ "$user" != "root" ]; then
```
Comment (tuminoid) [inline ci/dib_image/dib_elements/ubuntu-dev/post-install.d/02_configure:108]:
```suggestion
  if [[ "${user}" != "root" ]]; then
```

File: ci/dib_image/dib_elements/ubuntu-dev/post-install.d/02_configure [hunk 27]
```diff
@@ -0,0 +1,166 @@
...
+    sudo /usr/sbin/usermod -L "$user"
+    if [ "$user" != "sync" ] && [ "$user" != "shutdown" ] && [ "$user" != "halt" ]; then
```
Comment (tuminoid) [inline ci/dib_image/dib_elements/ubuntu-dev/post-install.d/02_configure:110]:
```suggestion
    if [[ "${user}" != "sync" ]] && [[ "${user}" != "shutdown" ]] && [[ "${user}" != "halt" ]]; then
```

File: ci/dib_image/dib_elements/ubuntu-dev/post-install.d/02_configure [hunk 28]
```diff
@@ -0,0 +1,166 @@
...
+      sudo /usr/sbin/usermod -s /sbin/nologin "$user"
```
Comment (tuminoid) [inline ci/dib_image/dib_elements/ubuntu-dev/post-install.d/02_configure:111]:
```suggestion
      sudo /usr/sbin/usermod -s /sbin/nologin "${user}"
```

File: ci/dib_image/dib_elements/ubuntu-dev/post-install.d/02_configure [hunk 29]
```diff
@@ -0,0 +1,166 @@
...
+sudo apt-get -y autoremove
+
+# We do not use passwords on the machines
+
+# Disable the system accounts for non-root users
+
+# shellcheck disable=SC2013
+for user in $(awk -F: '($3 < 500) {print $1 }' /etc/passwd); do
+  if [ "$user" != "root" ]; then
+    sudo /usr/sbin/usermod -L "$user"
```
Comment (tuminoid) [inline ci/dib_image/dib_elements/ubuntu-dev/post-install.d/02_configure:109]:
```suggestion
    sudo /usr/sbin/usermod -L "${user}"
```

File: ci/dib_image/openstack-vars.sh [hunk 30]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
```
Comment (tuminoid) [inline ci/dib_image/openstack-vars.sh:1]:
```suggestion
#!/usr/bin/env bash
```

File: scripts/update-nordix-repos-master.sh
Comment (tuminoid) [inline scripts/update-nordix-repos-master.sh:1]: This shouldn't be here, do a rebase.

**Review by tuminoid — COMMENTED** [review]
Nitting in this draft in hopes of catching the stuff early, so no more of the same is in the review version.

But its all nits, the actual work seems great.

**Comment by Sunnatillo** [comment]
/close
Disk Image builder workflow is implemented in [here](https://github.com/metal3-io/project-infra/pull/636)

**Pull request closed without merging**